contract/src/distribute/deposit.rs

```diff
@@ -0,0 +1,254 @@
+use aurora_launchpad_types::config::{DepositToken, TokenId};
+use near_sdk::json_types::U128;
+use near_sdk::{AccountId, Gas, Promise, PromiseResult, assert_one_yocto, env, near, require};
+
+use crate::traits::{ext_ft, ext_mt};
+use crate::{
+    AuroraLaunchpadContract, AuroraLaunchpadContractExt, GAS_FOR_FT_TRANSFER_CALL,
+    GAS_FOR_MT_TRANSFER_CALL, ONE_YOCTO,
+};
+
+const GAS_FOR_FINISH_ADMIN_WITHDRAW: Gas = Gas::from_tgas(10);
+
+#[near]
+impl AuroraLaunchpadContract {
+    #[payable]
+    pub fn distribute_deposit_tokens(&mut self) -> Promise {
+        assert_one_yocto();
+        require!(
+            self.is_success(),
+            "Deposited tokens could be distributed after success only"
+        );
+
+        require!(
+            !self.deposits_distribution.is_ongoing,
+            "Deposit distribution is ongoing"
+        );
+        self.deposits_distribution.is_ongoing = true;
+
+        let (solver_amount, fee_amount) = self
+            .calculate_distribution()
+            .unwrap_or_else(|e| env::panic_str(e));
+
+        require!(
+            solver_amount > 0 || fee_amount > 0,
+            "Deposit tokens have been already distributed"
+        );
+
+        match &self.config.deposit_token {
+            DepositToken::Nep141(token_account_id) => {
+                self.distribute_nep141_deposit_tokens(token_account_id, solver_amount, fee_amount)
+            }
+            DepositToken::Nep245((token_account_id, token_id)) => self
+                .distribute_nep245_deposit_tokens(
+                    token_account_id,
+                    token_id,
+                    solver_amount,
+                    fee_amount,
+                ),
+        }
+    }
+
+    fn distribute_nep141_deposit_tokens(
+        &self,
+        token_account_id: &AccountId,
+        solver_amount: u128,
+        fee_amount: u128,
+    ) -> Promise {
+        let solver_promise = (solver_amount > 0).then_some(
```

Comment:
```suggestion
        let solver_promise = (solver_amount > 0).then(
```

I think it makes sense to use `then` instead of `then_some` because the argument to the latter is always evaluated whereas the former takes a closure that is only evaluated when the bool is `true`.

This comment applies to all `then_some` calls in this file.